embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/telemetry/EmbraceTelemetryService.kt

```diff
@@ -50,6 +59,16 @@ internal class EmbraceTelemetryService(
         return result
     }
 
+    private fun getAndClearAppliedLimitTelemetry(): Map<String, String> {
+        synchronized(appliedLimitCountMap) {
```

Comment:
`synchronized` isn't really doing much here as this seems to be the only place where the synchronisation block occurs. I'd favour the approach used in `getAndClearStorageTelemetry` instead as it copies the map without requiring a mutex.